### PR TITLE
Read data using pandas

### PR DESCRIPTION
**Pull request by katybaulch**
# What's changed?

- Read data using pandas
- Update tests

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

File: gcf_data_mapper/read.py [hunk 1]
```diff
@@ -1,72 +1,165 @@
...
+        Pandas CSV reader. Otherwise this function will return None.
     """
-    with open(file_path, "r") as file:
-        csv_reader = csv.DictReader(file)
-        fieldnames = csv_reader.fieldnames or []
-        data = [{field: line[field] for field in fieldnames} for line in csv_reader]
-        return data
+    # Should the path exist, read the CSV contents into memory iteratively in chunks of
+    # 'chunk_size' (10**4 by default).
+    # This helps prevent out of memory errors where there is insufficient memory to
```
Comment (diversemix) [inline gcf_data_mapper/read.py:38]: An alternative (to using pandas) is to use a generator something like:
```
def _read_csvs(file_path1: str, file_path2: str) -> Generator:
    with open(file_path1) as f1, open(file_path2) as f2:
        reader1 = csv.DictReader(f1)
        reader2 = csv.DictReader(f2)
        for line1, line2 in zip(reader1, reader2):
            yield line1, line2
```
Not that one is more correct over the other, but this way does reduce dependencies :smile:

**Review by diversemix — APPROVED** [review]
Looks a good starting structure - let's get it in and iterate :fist_raised:

**Review by odrakes-cpr — APPROVED** [review]
great work on extending these functions

File: gcf_data_mapper/read.py (hunk 1 above)
Comment (katybaulch) [reply gcf_data_mapper/read.py:38]: Yeah, appreciate this :) I think pandas, despite introducing more dependencies, really does speed up the reading/retrieval of data (especially with larger datasets) and makes it a lot simpler to merge fields from files of different types :)

**Pull request merged**